### DISCUSSION TITLE
How to cut down the CPU consumption of Ltex?

### DISCUSSION THREAD

**Discussion by LittleFishLove** [valentjn/vscode-ltex#450] [Q&A] [answered]
I write my paper on vscode, and there are about 7k words. My machine is MacBook Pro (13-inch, 2020, Two Thunderbolt 3 ports), 16G, i5-8257U. When I input words, the ltex checks the words on a high CPU consumption.
![image](https://user-images.githubusercontent.com/35796964/140477370-dd3cc598-08b6-40ae-82f3-7a8719958943.png)
My setting is default and I make no changes.
I like the funtion of Ltex and enjoy writing paper in vscode, but the high CPU overhead degrades the inputting experience. Does anyone else meet the same problems? I'm so thirsty for a solution

**Comment by valentjn** [comment, marked as answer]
Did you read the [FAQ](https://valentjn.github.io/ltex/faq.html#why-does-ltex-have-such-a-high-cpu-load)? Have you tried increasing [`ltex.sentenceCacheSize`](https://valentjn.github.io/ltex/settings.html#ltexsentencecachesize)?

**Reply by LittleFishLove** [reply]
Yeah, I read it and  this https://github.com/valentjn/vscode-ltex/issues/134 and https://github.com/valentjn/vscode-ltex/issues/29; I set ltex.sentenceCacheSize as 1000, 2000, 4000, but the results have no difference.

**Reply by valentjn** [reply]
@LittleFishLove And the CPU consumption persists after the initial check (i.e., after the initial check, the CPU consumption increases again when typing)? Then it's probably a bug. Also considering we're getting lots of feedback in this direction lately, see #444 and Marketplace reviews. The problem is, it works perfectly fine for me, so I can't do anything about it without input.

Could you reproduce the issue with an example document that you're able to share and [open a bug report](https://valentjn.github.io/ltex/vscode-ltex/contributing.html#how-to-report-bugs) with the document attached (skip the minimal example step), and with the settings you use?

**Reply by LittleFishLove** [reply]
Fine, I will do that later, thank you!
